baseplate/clients/memcache/__init__.py

```diff
@@ -332,7 +332,7 @@ def _make_span(self, method_name: str) -> Span:
 
         """
         trace_name = f"{self.context_name}.{method_name}"
-        span = self.server_span.make_child(trace_name)
+        span = self.server_span.make_child(trace_name, component_name="memcache_client")
```

Comment:
I wonder if the `_client` part in all of these should be handed by `PrometheusClientSpanObserver` instead

Reply:
Right now, make_server_span goes through the server span observer where it looks at protocol and determines the metrics. Lass to use for that. In make_child it then treats them as client or local sounds (depending on an input var) but the metric names themselves come from component_name in the span.

When I used datadog, spans had component names too but it was more in reference to the mechanism used to collect the metrics than the metrics it was tracking (so you may have multiple request libs and differentiating the metrics from them would be via the component even if they're all http spans as an example). The thought here was to separate out the component name from the prefix from metrics to be able to identify that better, letting it come potentially from an extension of the client metrics class